tests/testthat/test-expect_identical_linter.R

```diff
@@ -0,0 +1,67 @@
+test_that("expect_identical_linter skips allowed usages", {
+  # expect_type doesn't have an inverted version
+  expect_lint("expect_true(identical(x, y) || identical(y, z))", NULL, expect_identical_linter())
+  # NB: also applies to tinytest, but it's sufficient to test testthat
+  expect_lint("testthat::expect_true(identical(x, y) || identical(y, z))", NULL, expect_identical_linter())
+
+  # expect_equal calls with explicit tolerance= are OK
+  expect_lint("expect_equal(x, y, tolerance = 1e-6)", NULL, expect_identical_linter())
+
+  # ditto for check.attributes = FALSE
+  expect_lint("expect_equal(x, y, check.attributes = FALSE)", NULL, expect_identical_linter())
+})
+
+test_that("expect_identical_linter blocks simple disallowed usages", {
+  expect_lint(
+    "expect_equal(x, y)",
+    rex::rex("expect_identical(x, y) is the default way of comparing two objects"),
+    expect_identical_linter()
+  )
+
+  # check.attributes = TRUE (set explicitly) gets dinged
+  expect_lint(
+    "expect_equal(x, y, check.attributes = TRUE)",
+    rex::rex("expect_identical(x, y) is the default way of comparing two objects"),
+    expect_identical_linter()
+  )
+
+  # different usage to redirect to expect_identical
+  expect_lint(
+    "expect_true(identical(x, y))",
+    rex::rex("expect_identical(x, y) is the default way of comparing two objects"),
+    expect_identical_linter()
+  )
+})
+
+test_that("expect_identical_linter skips cases likely testing numeric equality", {
+  expect_lint("expect_equal(x, 1.034)", NULL, expect_identical_linter())
+  expect_lint("expect_equal(x, c(1.01, 1.02))", NULL, expect_identical_linter())
+  # whole numbers with explicit decimals are OK, even in mixed scenarios
+  expect_lint("expect_equal(x, c(1.0, 2))", NULL, expect_identical_linter())
+  # plain numbers are still caught, however
+  expect_lint(
+    "expect_equal(x, 1L)",
+    rex::rex("expect_identical(x, y) is the default way of comparing two objects"),
+    expect_identical_linter()
+  )
+  expect_lint(
+    "expect_equal(x, 1)",
+    rex::rex("expect_identical(x, y) is the default way of comparing two objects"),
+    expect_identical_linter()
+  )
+  expect_lint(
+    "expect_equal(x, TRUE)",
+    rex::rex("expect_identical(x, y) is the default way of comparing two objects"),
```

Comment:
NB this would send a lint after refactoring for not using `expect_true(x)`.

Reply:
I'll note that in a comment... this test is to make sure we get each of the different flavors of `NUM_CONST`